cmake/linker/ld/target.cmake

```diff
@@ -137,14 +126,44 @@ function(toolchain_ld_link_elf)
     ${LINKERFLAGPREFIX},--no-whole-archive
     ${NO_WHOLE_ARCHIVE_LIBS}
     $<TARGET_OBJECTS:${OFFSETS_LIB}>
-    ${LIB_INCLUDE_DIR}
     -L${PROJECT_BINARY_DIR}
-    ${TOOLCHAIN_LIBS}
```

Comment:
Now `-lhal` is missing when using `xt-xcc` and compilation fails with undefined `xthal_window_spill` (`xthal_window_spill` is provided by the `xt-xcc` toolchain in `libhal.a`)

If I restore this specific line then `xt-xcc` compilation succeeds again.

I wonder why was this `${TOOLCHAIN_LIBS}` line removed from SOME `target.cmake` files by commit 2e3873adde2 but as of today's commit 27456ed2b57f it is still present in the following files? This seems inconsistent.

```
cmake/compiler/icx/target.cmake
cmake/linker/arcmwdt/target.cmake
cmake/linker/xt-ld/target.cmake
```

Indeed, if I remove `${TOOLCHAIN_LIBS}` from `cmake/linker/xt-ld/target.cmake` then `xt-clang` fails with the same error.

cc: @kv2019i , @andyross, @cujomalainey, @dcpleung 

FYI:

```
cmake/compiler/xcc/target.cmake:list(APPEND TOOLCHAIN_LIBS
cmake/compiler/xcc/target.cmake-  gcc
cmake/compiler/xcc/target.cmake-  hal
cmake/compiler/xcc/target.cmake-  )
```

Reply:
an oversight.

I initially wanted to still allow for the old use of `TOOLCHAIN_LIBS` for toolchains I didn't have available, but later changes related to picolibc has actually resulted in the situation that all toolchains must be updated.

Reply:
Thanks @tejlmand !

> actually resulted in the situation that all toolchains must be updated.

While @dcpleung is removing the obsolete TOOLCHAIN_LIBS for Xtensa in #79731, as of today I see it still present in the main branch for `cmake/compiler/icx/target.cmake` and `cmake/linker/arcmwdt/target.cmake`. I think it would be good to file a bug and use `git blame` to warn the owners of these files that TOOLCHAIN_LIBS does not work anymore in case they still try and fail to use it downstream somehow. Something I did not mention before: it took me HOURS to debug this Xtensa compilation failure and find that it was down to this TOOLCHAIN_LIBS variable. Funny enough, `git grep hal` does not work very well! It would be nice to avoid others' a similar... effort.

In case these toolchains have no owner left, then it is a great opportunity to notice that and delete them entirely.

There is also a mysterious `TOOLCHAIN_LIBS_OBJECTS` in `cmake/linker/armlink/target.cmake`... My 2 cents.